src/KingSystem/Damage/dmgDamageManagerBase.cpp

```diff
@@ -0,0 +1,271 @@
+    paramsList = mActor->mActorParam->gParamList;
+
+    const ksys::res::GParamListObjectGeneral& params = paramsList->getGeneral();
+    if (params.mIsLifeInfinite.ref()) {
+        // Since life is infinite, we don't need to modify the damage or life.
+        // But we still call the "callback" as if damage was done.
+        onApplyDamage();
+        return false;
+    }
+
+    s32 damage = getDamage();
+    if (damage >= 1) {
+        if (isPlayerClass(mActor)) {
+            if (getAttacker()->hasProc()) {
+                tryBuffDamage(damage);
+            }
+        }
+    }
+
+    tryApplyDamageRecovery(damage);
+
+    // Actually deal damage
+    onApplyDamage();
+    f32 oldLife = life;  // Random float for some reason
+    // Force positive result(Any negative result right shifted by 31 = -1, positive result = 0)
+    life = (life - damage) & ~((life - damage) >> 0x1F);
+    bool didDamage = oldLife != life;
+
+    return didDamage;
+}
+
+void DamageManagerBase::tryBuffDamage(s32& damage) {
+    if (!uking::aoc2::instance()) {
+        return;
+    }
+
+    if (!uking::aoc2::instance()->checkFlag2(uking::aoc2::Flags2::EnableHardMode)) {
+        return;
+    }
+
+    if (!uking::aoc2::instance()->checkFlag1(uking::aoc2::Flags1::ApplyDamageMultiplier)) {
+        return;
+    }
+
+    ksys::act::ActorConstDataAccess acc;
+    ksys::act::acquireActor(getAttacker(), &acc);
+    if (uking::aoc2::shouldApplyMasterModeDamageMultiplier(acc)) {
+        uking::aoc2::buffDamage(damage);
+    }
+}
+
+void DamageManagerBase::tryApplyDamageRecovery(s32& damage) {
+    if (!uking::aoc2::instance()) {
+        return;
+    }
+
+    if (!uking::aoc2::instance()->checkFlag2(uking::aoc2::Flags2::EnableHardMode)) {
+        return;
+    }
+
+    if (!uking::aoc2::instance()->checkFlag1(uking::aoc2::Flags1::EnableLifeRegen)) {
+        return;
+    }
+
+    if (!mActor->getLifeRecoverInfo()) {
+        return;
+    }
+
+    // Take damage from extra HP1, and modify damage.
+    // Returns true if damage remains? No more regen?
+    ksys::act::LifeRecoverInfo* lifeRecoveryInfo = mActor->getLifeRecoverInfo();
+    if (lifeRecoveryInfo->onApplyDamage(damage)) {
+        // Update flags and counter.
+        lifeRecoveryInfo->onApplyDamage_0();
+    }
+}
+
+s32 DamageManagerBase::getNumCallbacks() {
+    // More like number of supported event types, than number of callbacks.
+    // These seem to return a fixed number for each different DamageManager class.
+    return 0;
+}
+
+bool DamageManagerBase::initCallbacks(sead::ExpHeap*) {
+    // Does nothing in the Base class, and doesn't seem like any of the known classes actually
+    // override this yet.
+    return true;
+}
+
+bool DamageManagerBase::allocStruct20(sead::ExpHeap* heap) {
+    mStruct20_a = new (heap, 8) Struct20;
+    if (!mStruct20_a) {
+        return false;
+    }
+
+    mStruct20_b = new (heap, 8) Struct20;
+    return mStruct20_b != NULL;
```

Comment:
```suggestion
    return mStruct20_b != nullptr;
```

nullptr is preferred since C++11.